extensions/data-transfer/portability-data-transfer-microsoft/src/test/java/org/dataportabilityproject/transfer/microsoft/integration/LocalImportTestRunner.java

```diff
@@ -40,20 +35,13 @@
     public static void main(String... args) throws Exception {
         AuthTestDriver authTestDriver = new AuthTestDriver();
 
-        OkHttpClient client = new OkHttpClient.Builder().build();
-
-        ObjectMapper mapper = new ObjectMapper();
-
-        TransformerService transformerService = new TransformerServiceImpl();
-
-        MicrosoftTransferServiceProvider serviceProvider = new MicrosoftTransferServiceProvider(client, mapper, transformerService);
+        MicrosoftTransferServiceProvider serviceProvider = new MicrosoftTransferServiceProvider();
```

Comment:
My plan was not to use Guice for the MS extensions. The thinking here is to minimize the knowledge/libraries needed to author an extension, and allow for people to opt-in to using a wider array of libraries if they choose. The MS extensions can serve as a way to make sure it is possible to write these types of "minimalist" extensions.

The AbstractExtension implementations for the MS extensions will compose the required services manually so that's why I have the ctor params. This is also needed for testing to pass in configured or mock instances.

Reply:
@jimmarino I had done it because I didn't want to introduce MS-specific dependencies like OkHttpClient into the core worker lib, assuming the worker will instantiate the MS TransferServiceProvider and thus need to pass in those vars to construct it. Do you have any thoughts on how the TransferServiceProviders can be loaded, ideally without introducing those deps into the worker?

Reply:
I like the idea of leaving the constructor take the params that are needed for a service provider, with or without Guice, for wiring up for tests and binaries.

For this specific dep, I think OkHttpClient (or eventually a wrapper around http fetching) is common enough that it would be needed outside of MS.

I think, per Rachel's question, we need to figure out how we satisfy dependencies when it gets bootstrapped in a binary.

Reply:
Ah OK I see the problem. I think the model has to be that the extensions provide specific service instances such as TransferServiceProvider as opposed to the core creating them.

There is a TransferExtension (subtype of AbstractExtension) for doing that. An extension provides a TransferExtension implementation that wires up extension services and provides the instances to the bootstrapper using the JAR service loader mechanism.  There is a problem currently that TransferExtension has getImporters() and getExporters() but maybe it should be getTransferServiceProvider()? In the future, perhaps we could look to eliminate TransferServiceProvider if it is not needed. 

WorkerMain has the start of this model. There may be some timing issues to deal with, for example, the ObjectMapper will need to be created before extensions are installed and bound into the Guice injector afterward. After the extensions are loaded, their returned instances can be bound into the Guice injector so they are available for wiring in the core.

Let me know if this will work or a different approach is needed.